src/Simplify_Min.cpp

```diff
@@ -96,6 +98,7 @@ Expr Simplify::visit(const Min *op, ExprInfo *bounds) {
                rewrite(min(x, ((x + c0)/c1)*c1 + c2), a, c1 > 0 && c0 + c2 >= c1 - 1) ||
                rewrite(min(((x + c0)/c1)*c1 + c2, x), a, c1 > 0 && c0 + c2 <= 0) ||
                rewrite(min(x, ((x + c0)/c1)*c1 + c2), b, c1 > 0 && c0 + c2 <= 0) ||
+               rewrite(min((x/c0)*c0, (x/c1)*c1 + c2), a, c2 >= c1 && c1 > 0) ||
```

Comment:
Is this true? What if `x` is negative?

Reply:
I think:  x = -2, c0 = 0, c1 = 1, and c2 = 1 is a counterexample, produced by z3.

Reply:
I think x negative is fine, but it looks like this needs a condition that c0 > 0

Reply:
Yeah, that works for z3

Reply:
This verified as correct for me... there's an implicit assumption you can make that c0 != 0. Is that enough?

Reply:
The LHS is always less than x, even for negative c0, and the RHS is always greater than x

Reply:
`c0 != 0` also works for my verifier. @abadams are you using the super_simplifier stuff? I have a python translator that's just easy for me to use, but it doesn't seem to understand the implicit assumption c0 != 0, where does that come from?

Reply:
https://github.com/halide/Halide/blob/super_simplify_v2/apps/super_simplify/filter_rewrite_rules.cpp#L215

Reply:
Oops okay maybe I should use that code instead. Is this predicate implicit simply because it should have been removed by an earlier rewrite rule already?

Reply:
Yeah, we can assume constant folding and canonicalization rules (e.g. in x + y, x is not a constant) run before more general rules.